### ISSUE TITLE
Buttons + icons not vertically aligned

### ISSUE THREAD

**Issue by GianlucaGuarini** [google/material-design-lite#4165]
Is there a way to get the icons in the buttons vertically aligned?
As you can see from my example all the icons are not vertically aligned with the text.

![screen shot](https://cloud.githubusercontent.com/assets/879612/13420868/0820106c-df8a-11e5-86a8-67699d6bf3ad.png)

[Demo](https://jsfiddle.net/gianlucaguarini/ebhctb6m/)

Please let me know if you have any plan to update this behavior. I will be glad to submit a pull request

**Comment by GianlucaGuarini** [comment]
I think at moment it's only possible with weird css hacks (and this sucks!).

![screen shot 1](https://cloud.githubusercontent.com/assets/879612/13421497/0601fbe2-df8f-11e5-9d0d-b3976ca44dac.png)

https://jsfiddle.net/gianlucaguarini/7ntmanxm/

**Comment by Zodiase** [comment]
You could use flexbox inside `mdl-button`.

**Comment by GianlucaGuarini** [comment]
It does not solve the problem :(

**Comment by Zodiase** [comment]
@GianlucaGuarini Oh really? http://codepen.io/Zodiase/pen/MyYxjJ

**Comment by Zodiase** [comment]
Alternatively, since the `line-height` of `mdl-button` is set based on `16px` height content, you could shrink the icon from `24px` to `16px`, or offset it as you are doing right now.

**Comment by GianlucaGuarini** [comment]
@Zodiase should not this be supported directly by mdl?

**Comment by GianlucaGuarini** [comment]
@Zodiase btw have you tested your demo on firefox?

**Comment by Zodiase** [comment]
@GianlucaGuarini Nope I mainly develop for Chrome.

**Comment by Zodiase** [comment]
@GianlucaGuarini Demo updated to work for Firefox and Safari.

**Comment by GianlucaGuarini** [comment]
thanks @Zodiase but that's the result of the demo ( i have also tried other kind of icons):

Firefox
![screen shot 2](https://cloud.githubusercontent.com/assets/879612/13453181/4811130a-e04e-11e5-896b-60bc68f9bfac.png)

Safari
![screen shot 3](https://cloud.githubusercontent.com/assets/879612/13453249/d1cf3bd0-e04e-11e5-8869-8dd6d7b0fc56.png)

Chrome
![screen shot 4](https://cloud.githubusercontent.com/assets/879612/13453267/f347cdb8-e04e-11e5-865b-24f4fa5f9cc0.png)
![screen shot 5](https://cloud.githubusercontent.com/assets/879612/13453285/0edcdc8a-e04f-11e5-9c46-cb832642ff1e.png)
![screen shot](https://cloud.githubusercontent.com/assets/879612/13453318/5a7106da-e04f-11e5-84b4-39550f549805.png)

It seems the icons are always 1px misaligned from the top. I am wondering whether we need an official solution and not trying to hack the issue. I guess also the lateral padding is not nice, I think a button with an icon should look more like [my previous example](https://github.com/google/material-design-lite/issues/4165#issuecomment-190609222)
Of course your solution is ok but it does not produce any better result compared to the default icon + button solution ( without flexbox ). I think this topic is quite important if we want to use __mdl__ for building professional pixel perfect UIs.. for amateurs the current solution is more than ok.

**Comment by Garbee** [comment]
We need MD Guidance here to see if icons in buttons are acceptable. Currently the specification does not say anything about the occurrence.

**Comment by sgomes** [comment]
@GianlucaGuarini: @Garbee is correct in that icons inside rectangular buttons aren't a documented use-case in the spec. I'm reluctant for us to ship fixes to improve this use-case without any guidance into how to do it correctly (if it is indeed correct usage); I'd want to know whether it would affect button size, what the paddings should be like, etc.

**Comment by GianlucaGuarini** [comment]
As side note, the buttons + icons are already used on https://www.polymer-project.org/1.0/ having an awful design ( wrong padding and alignment )

![screen shot](https://cloud.githubusercontent.com/assets/879612/13662502/491abce4-e69b-11e5-8166-7d788dd39580.png)

So probably It's worth spending some time defining the specs regarding how material should handle these kind of elements

cc @ebidel

**Comment by rejhgadellaa** [comment]
Looks like the problem is not so much the icon as it is the (bottom space) of the text? If text is aligned vertically it sits a bit higher to accommodate for the lower parts of letters like 'g', 'y', etc.

**Comment by ebidel** [comment]
@GianlucaGuarini The existing Polymer site uses polymer 0.5, which uses the old element set. However, the text seems vertically centered to icon to me. The same is true with the 1.0 set: https://elements.polymer-project.org/elements/paper-button?view=demo:demo/index.html&active=paper-button.

AFAIK, the icons themselves no not build in the padding. This is left to other elements like paper-icon-button and paper-item.

Keep in mind that one thing that can affect alignment is line-height. if the page has set this, than it will inherit and can mess things up.

**Comment by GianlucaGuarini** [comment]
@ebidel you are right the icons look vertically aligned but only on chrome, even on the last paper-icon-button version there seems to be 1px misalignment at least on firefox (48.0a2 Macbook Pro 10.11.4).

![screen shot](https://cloud.githubusercontent.com/assets/879612/14910554/af04f248-0dee-11e6-90e9-342e54decc82.png)

> Keep in mind that one thing that can affect alignment is line-height. if the page has set this, than it will inherit and can mess things up.

Yes of course, I agree.. but here I am speaking about a vanilla version of mdl that should already provide a solution to these problems IMHO

**Comment by ebidel** [comment]
Yes but what I'm saying is that line-height is an inheritable property. So
if the page has messed with it, it's possible it will affect the pre-made
solution. This is one reason we need shadow dom so badly.

**Comment by broonage** [comment]
I still have the same issue, the icons are not aligned with the text
